pkg/config/resource.go

```diff
@@ -160,8 +160,12 @@ type Resource struct {
 	// to overwrite it.
 	IDFieldName string
 
-	// Group is the group of CRD.
-	Group string
+	// ShortGroup is the short name of the API group of this CRD. The full
+	// CRD API group is calculated by adding the group suffix of the provider.
+	// For example, ShortGroup could be `ec2` where group suffix of the
+	// provider is `aws.crossplane.io` and in that case, the full group would
+	// be `ec2.aws.crossplane.io`
+	ShortGroup string
```

Comment:
`ShortGroup` is fine but we can also consider something like `GroupPrefix` as its name. From a "Short" version, I expect something that _substitutes_ a longer version of it, like in the context of command-line options, "-d" being a substitute for "--debug".

Reply:
Yes, in fact, it substitutes the longer version, i.e. `ec2` for `ec2.aws.crossplane.io`. But I don't love `ShortGroup` either. `Group` only is misleading because the actual group is `ec2.aws.crossplane.io`.

The main motivation was that I didn't want to treat those two fields (`GroupPrefix` and `GroupSuffix`) as pure strings with those names because the concatenation we're doing now is not simply prepending anymore; we're opening  a new group under the root group if short group is given. Open to suggestions for the name.